float64[3] position
float64[3] velocity
float64 pos_var